assets/events.py

```diff
@@ -0,0 +1,203 @@
+        log.debug('environment: %s', os.environ)
+
+
+    def check_cmd(self, source, version):
+        """
+        Check for new version(s)
+
+        #TODO this first release of the resource won't check for new events/won't trigger
+        But in a future version, we may want to check/filter specific events to trigger the pipeline
+
+        :param source: is an arbitrary JSON object which specifies the location of the resource,
+        including any credentials. This is passed verbatim from the pipeline configuration.
+        :param version: is a JSON object with string fields, used to uniquely identify an instance of the resource.
+        :return: a dict with the version fetched
+        """
+
+        log.debug('version: %s', version)
+
+        if not version:
+            version = {"timestamp": '0'}
+        log.debug('source: %s', source)
+        log.debug('version: %s', version)
+
+        return versions_list
+
+
+    def in_cmd(self, target_dir, source, version, params):
+        """
+        Get an event from Cycloid API
+
+        #TODO this first release of the resource won't get an event, we don't have usecase right now
+
+        :param target_dir: a temporary directory which will be exposed as an output
+        :param source: is the same value as passed to check
+        :param version: is the same type of value passed to check, and specifies the version to fetch.
+        :param params: is an arbitrary JSON object passed along verbatim from params on a get.
+        :return: a dict with the version fetched and the metadata
+        """
+        log.debug('source: %s', source)
+        log.debug('version: %s', version)
+
+        if not version:
+            version = {'timestamp': '0'}
+
+        metadata = []
+
+        metadata_path = os.path.join(target_dir, 'metadata')
+        with open(metadata_path, 'w') as metadata_file:
+            json.dump(metadata, metadata_file)
+
+        return {
+            'version': {'timestamp': '0'},
+            'metadata': metadata,
+        }
+
+    def out_cmd(self, target_dir, source, params):
+        """
+
+        :param target_dir:
+        :param source: is the same value as passed to check.
+        :param params: is an arbitrary JSON object passed along verbatim from params on a put.
+        :return: a dict with the version fetched and the metadata
+        """
+
+        merge = self._merge_source_params(source, params)
+        self._check_params('icon', merge)
+        self._check_params('message', merge)
+        self._check_params('severity', merge)
+        self._check_params('title', merge)
+        self._check_params('type', merge)
+        self._check_params('tags', merge)
+
+        self._send_events()
+
+        metadata = []
+
+        metadata_path = os.path.join(target_dir, 'metadata')
+        with open(metadata_path, 'w') as metadata_file:
+            json.dump(metadata, metadata_file)
+
+        return {
+            'version': {'timestamp': '0'},
```

Comment:
Is it always this version?

Reply:
We don't want to have versions yet, as we don't have checks. Version my be a unique id of logs in the future, I don't know yet. But for just a "PUT", we don't need versions of the resource, so a static one will avoid multiple versions like the docurated/vault resource is doing and pollute the database.